tamr_client/dataset/dataframe.py

```diff
@@ -7,9 +7,11 @@
 
 import pandas as pd
 
-import tamr_client as tc
 from tamr_client.types import JsonDict
-
+from tamr_client.dataset import Dataset
```

Comment:
To keep internal `import`s from depending on `__init__.py` hoisting, this should be:

```suggestion
from tamr_client.dataset.dataset import Dataset
```

(Note: this repetition is fine. we can do a follow-on PR where we split up the data class and functions and rename modules where the `dataset.dataset` repetition will be addressed)